File: hourglass.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'hourglassSum' function below.
#
# The function is expected to return an INTEGER.
# The function accepts 2D_INTEGER_ARRAY arr as parameter.
#

# See https://www.hackerrank.com/challenges/2d-array/problem for details

def hourglassSum(arr):
    sums = []
    
    for i in range(1, len(arr) - 1):
        for j in range(1, len(arr) - 1):
            topSection = arr[i-1][j-1] + arr[i-1][j] + arr[i-1][j+1]
            midSection = arr[i][j]
            bottomSection = arr[i+1][j-1] + arr[i+1][j] + arr[i+1][j+1]
            currentSum = topSection + midSection + bottomSection
            sums.append(currentSum)
    return max(sums)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    arr = []

    for _ in range(6):
        arr.append(list(map(int, input().rstrip().split())))

    result = hourglassSum(arr)

    fptr.write(str(result) + '\n')

    fptr.close()

File: hourglass.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'hourglassSum' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts 2D_INTEGER_ARRAY arr as parameter.
 */

function hourglassSum(arr) {
    let sums = []
    
    for (let i = 1; i < arr.length - 1; i++) {
        for (let j = 1; j < arr.length - 1; j++) {
            let topSection = arr[i-1][j-1] + arr[i-1][j] + arr[i-1][j+1]
            let midSection = arr[i][j]
            let bottomSection = arr[i+1][j-1] + arr[i+1][j] + arr[i+1][j+1]
            let currentSum = topSection + midSection + bottomSection
            sums.push(currentSum)
        }
    }
    return Math.max(...sums)
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    let arr = Array(6);

    for (let i = 0; i < 6; i++) {
        arr[i] = readLine().replace(/\s+$/g, '').split(' ').map(arrTemp => parseInt(arrTemp, 10));
    }

    const result = hourglassSum(arr);

    ws.write(result + '\n');

    ws.end();
}
